internal/manifests/targetallocator/service.go

```diff
@@ -0,0 +1,43 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package targetallocator
+
+import (
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/aws/amazon-cloudwatch-agent-operator/internal/manifests"
+	"github.com/aws/amazon-cloudwatch-agent-operator/internal/naming"
+)
+
+func Service(params manifests.Params) *corev1.Service {
+	version := strings.Split(params.OtelCol.Spec.TargetAllocator.Image, ":")
+	labels := Labels(params.OtelCol, naming.TargetAllocatorServiceName)
+	if len(version) > 1 {
+		labels["app.kubernetes.io/version"] = version[len(version)-1]
+	} else {
+		labels["app.kubernetes.io/version"] = "latest"
+	}
+
+	selector := Labels(params.OtelCol, naming.TargetAllocatorServiceName)
+
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      naming.TargetAllocatorServiceName,
```

Comment:
I think @okankoAMZ mentioned this needs to be hardcoded

Reply:
Nvm, its updated here - https://github.com/aws/amazon-cloudwatch-agent-operator/blob/557c263b3ec435e832f68a86531647b2aede89b8/internal/naming/main.go#L7